{"pageProps":{"meta":{"title":"Python3 Protobuf Import Error","date":"2017/01/02"},"content":"<h1 id=\"python3error\">Python 3 error</h1>\n<p>I'm trying to make the switch to Python 3 at the moment, and was getting incredibly frustrated getting this error when I imported a module using Google's Protocol Buffers:</p>\n<pre><code class=\"python language-python\">Traceback (most recent call last):\n  File \"/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/google/pr\notobuf/internal/python_message.py\", line 1087, in MergeFromString\n    if self._InternalParse(serialized, 0, length) != length:\n  File \"/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/google/pr\notobuf/internal/python_message.py\", line 1109, in InternalParse\n    (tag_bytes, new_pos) = local_ReadTag(buffer, pos)\n  File \"/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/google/pr\notobuf/internal/decoder.py\", line 181, in ReadTag\n    while six.indexbytes(buffer, pos) &amp; 0x80:\nTypeError: unsupported operand type(s) for &amp;: 'str' and 'int'\n</code></pre>\n<p>After much messing around and not finding a fix from google, it turns out I had the wrong version of protobuf installed. Very easy fix:</p>\n<pre><code>pip3 uninstall protobuf &amp;&amp; pip3 install python3-protobuf\n</code></pre>\n<p>Hopefully this post helps someone solve the same problem.</p>"},"__N_SSG":true}